src/proto/flwr/proto/appio.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package flwr.proto;
+
+import "flwr/proto/fab.proto";
+import "flwr/proto/run.proto";
+import "flwr/proto/transport.proto";
+
+service ClientAppIo {
+  // Get Message, Context, FAB, and Run
+  rpc PullClientAppInputs(PullClientAppInputsRequest)
+      returns (PullClientAppInputsResponse) {}
+
+  // Send updated Message and Context
+  rpc PushClientAppOutputs(PushClientAppOutputsRequest)
+      returns (PushClientAppOutputsResponse) {}
+}
+
+message PullClientAppInputsRequest { sint64 token = 1; }
+message PullClientAppInputsResponse {
+  Message message = 1;
+  Context context = 2;
+  Run run = 3;
+  // Fab fab = 4;
+}
+message PushClientAppOutputsRequest {
+  sint64 token = 1;
+  Message message = 2;
+  Context context = 3;
+}
+message PushClientAppOutputsResponse { Status status = 1; }
```

Comment:
We should probably use a different `Status` here (not the one imported from `transport.proto`)

Reply:
Good point. I've added the `SUCCESS`, `DEADLINE_EXCEEDED` and `UNKNOWN_ERROR` to the `Code` message type. I think that these should be sufficient for now and we can be more specific later. What do you think @danieljanes?

Reply:
I wonder if we could match the error codes to this [ErrorCode](https://github.com/adap/flower/blob/3034810c525e175cfce95fcad70dd4f23fd4934a/src/py/flwr/common/constant.py#L101) or expand it if necessary.

Reply:
@jafermarq Good suggestion, but since those are for error codes, I'll leave the refactoring of the codes to another PR.